src/uffd_handler/src/prefaulter.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::prefaulter::PrefaulterError::InvalidAmount;
+
+const KIB_MULT: usize = 1 << 10;
+
+#[derive(Debug, thiserror::Error)]
+pub enum PrefaulterError {
+    #[error("Amount to prefault in bytes {0} the faulting address exceeds usize: {1}")]
+    InvalidAmount(String, usize),
+}
+
+#[derive(Default)]
+pub struct UffdPrefaulter {
+    bytes_before: usize,
+    bytes_after: usize,
+}
+
+impl UffdPrefaulter {
+    pub fn new(before: usize, after: usize) -> Result<Self, PrefaulterError> {
+        Ok(UffdPrefaulter {
+            bytes_before: before
+                .checked_add(KIB_MULT)
```

Comment:
Is this the check that verifies that the number of KiB provided for before/after prefaulting is not equivalent to a number of bytes that is still `usize` (and doesn't produce an overflow)? I mean, I'm not quite sure why is `.checked_add(KIB_MULT)` and not `.checked_mul(KIB_MULT)` in order to convert KiB (as parsed from cmdline) to bytes (as saved in the `struct UffdPrefaulter`)

Reply:
Ah.. so you fixed that in the next commit (feature: uffd: implement configurable prefault). Sorry, I wasn't reading until the end before writing. However, it would be nice to change the second commit, also (feature: uffd: add optional arguments )